meson.build

```diff
@@ -2,9 +2,8 @@
 project(
     'dinit', 
     'cpp', 
-    version : run_command(
-        ['sh', 'build/general_variables_for_meson.sh', 'version'],
-        check : true).stdout(),
+    version : run_command('grep', 'VERSION', 'build/version.conf',
```

Comment:
This technique is fragile. I think it should at least:

-  look for `VERSION=` instead of just `VERSION` (and similar for other variables). And require that the string is found at the beginning of the line.  This is more robust in case new variables are added later that begin with or contain the same letters.
- use the calculated length of `VERSION=` instead of hardcoding the value (8). Though, meson doesn't seem to have a documented way to determine the length of a string. Perhaps `len('VERSION=')` will work. Reason: if it is hardcoded it is easy to miss updating it in case the name changes.

Reply:
What about `.stdout().strip().split('=')[1]`? Split it into an array on each equals sign, and take the second component. You know there will only ever be two components so you don't need to assert that.

Reply:
I replaced `.substring()` with `.split('=')` for more robust variable definition. it's fix hardcoded `.substring()` for finding variable values.
But
> And require that the string is found at the beginning of the line.

Hmm, About this I didn't found a robust solution. but I found some ideas:
1. `grep` support ending up finding paterns via `-m` flag after found n matching numbers of line. for example if `VERSION=` before `TEST_VERSION=`, we can use `grep -m1 VERSION= build/version.conf`.

2. meson have `keyval` module designed for reading these files. it's robustest way to do that but modules in meson can't be called before `project()` function and I think it means we can't use it for version. for example
    ```meson
    ...
    meson_version : '>= 0.56.0'
    ...

    ## Import variables from build/version.conf
    kv = import('keyval')
    version_conf = kv.import('build/version.conf`)

   ## General defines
   ...
   month = version_conf.get('MONTH')
   year = version_conf.get('YEAR')
   ...
   ```
Regards

Reply:
>  >   And require that the string is found at the beginning of the line.

> Hmm, About this I didn't found a robust solution

How about "grep ^VERSION= build/version.conf"?